src/base/LruMap.h

```diff
@@ -43,7 +56,7 @@ template <class EntryValue, size_t EntryCost = sizeof(EntryValue)> class LruMap
     /// Search for an entry, and return a pointer
     EntryValue *get(const char *key);
     /// Add an entry to the map
-    bool add(const char *key, EntryValue *t);
+    bool add(const char *key, EntryValue *t, int ttl);
```

Comment:
I response to your #2. The first thing I looked at was having them use the original creation value. defaultTtl() is the result of two problems with that approach.  Several of the callers of add() do not have access to the original information used to generate the LruMap, and worse that initialization value is overloading multiple concepts - disabled caching, TTL based removal, and LRU-only storage. The defaultTtl() accessor is for extracting the LruMap's initial/default concept of TTL (values 0 thru +N) out of the retention member - not using all its entire range of content values.

I can accept this latest proposal, but it will need to do that same mapping defaultTtl() does to avoid negative retention values (infinite freshness) being confused with already-stale negative TTL values.